From 99f06cbe34647a48f881ef16a50c804b3567fe90 Mon Sep 17 00:00:00 2001
From: Xie XiuQi <xiexiuqi@huawei.com>
Date: Fri, 26 Feb 2021 20:21:12 +0800
Subject: [PATCH] arm64/mpam: enable alloc/mon capable when MPAM enabled
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 99f06cbe34647a48f881ef16a50c804b3567fe90


hulk inclusion
category: feature
feature: ARM MPAM support
bugzilla: 48265
CVE: NA

--------------------------------

Signed-off-by: Xie XiuQi <xiexiuqi@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Signed-off-by: Wang ShaoBo <bobo.shaobowang@huawei.com>
Reviewed-by: Cheng Jian <cj.chengjian@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm64/kernel/mpam.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm64/kernel/mpam.c b/arch/arm64/kernel/mpam.c
index 9cc9b2a232a9..260490dad226 100644
--- a/arch/arm64/kernel/mpam.c
+++ b/arch/arm64/kernel/mpam.c
@@ -1283,6 +1283,9 @@ static int __init mpam_late_init(void)
 	if (!cpus_have_const_cap(ARM64_HAS_MPAM))
 		return -ENODEV;
 
+	rdt_alloc_capable = 1;
+	rdt_mon_capable = 1;
+
 	mpam_init_padding();
 
 	ret = mpam_nodes_init();
-- 
2.26.2

